lib/worker-manager.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+const child_process = require('child_process');
+const events = require('events');
+const debug = require('debug')('brunch:workers');
+
+const workerFile = `${__dirname}/workeryo.js`;
+
+const genId = (() => {
+  let counter = 0;
+
+  return () => counter++;
+})();
+
+class WorkerManager {
+  constructor(persistent, options, config) {
+    this.jobs = [];
+    this.workers = [];
+    this.pending = {};
+    this.free = [];
+    this.events = new events;
+    this.options = options;
+    this.config = config;
+
+    const num = typeof options.parallel === 'string' ? parseInt(options.parallel) : require('os').cpus().length;
```

Comment:
Using `os.cpus.length` is not always reliable. On some newer macbooks, OS X reports double the original number of cores because "hyper threading", however, I've found that 4 workers are actually way slower than 2 on my macbook. There doesn't seem to be a way to get the real number of cores, so it's better to always pass `-J num`